File: secant_method.py
from sympy import *
x=symbols('x')




def secant_method(expr,x0,x1,steps):
    x=symbols('x')
    f=sympify(expr)
    xn=x1
    xn_minus_1=x0


    print"\n\n"
    print"#             x"
    print 0,"      ",'%.8f'% (x0)
    print 1,"      ",'%.8f'% (x1)
    for i in range (2,steps+1):
    
        fxn=f.subs(x,xn)
        fxn_minus_1=f.subs(x,xn_minus_1)
        xn_plus_1=xn-fxn*(xn-xn_minus_1)/(fxn-fxn_minus_1)
    
        print i,"      ",'%.8f'% (xn_plus_1)
    
        xn_minus_1=xn
        xn=xn_plus_1
    


func=str(input("Provide a function: "))
s=int(input("How many iterations?: "))
b=float(input("Provide an initial upper guess: "))
a=float(input("And an initial lower guess: "))


secant_method(func,a,b,s)

File: secant_method.m
clear
syms x;


f=x^2-5;
x0=2;
x1=3;
steps=4;


xn=x1;
xn_minus_1=x0;


fprintf("\n\n")
fprintf("#           x\n")
fprintf("%d      %.8f\n",0,x0)
fprintf("%d      %.8f\n",1,x1)
for i=2:steps
    
    fxn=subs(f,x,xn);
    fxn_minus_1=subs(f,x,xn_minus_1);
    xn_plus_1=xn-fxn*(xn-xn_minus_1)/(fxn-fxn_minus_1);
    
    fprintf("%d      %.8f\n",i,xn_plus_1)
    
    xn_minus_1=xn;
    xn=xn_plus_1;
    
end